### ISSUE TITLE
Indentation level is removed when a function call has lambda expression in one parameter

### ISSUE THREAD

**Issue by dimsuz** [pinterest/ktlint#1221]
## Observed Behavior

Formatting this (correctly formatted) code with ktlint
```kotlin
class MoneyValidatorsTest : ShouldSpec({
  context("isMoneyValue") {
    should("not error on strings with integers") {
      checkAll( Arb.int().map { it.toString() }, Arb.currency()) { value, currency ->
        val sut = isMoneyValue(currency)

        val result = sut.validate(value)

        result shouldBe Ok(Money(value, currency))
      }
    }
  }
}
```
results in
```kotlin
class MoneyValidatorsTest : ShouldSpec({
  context("isMoneyValue") {
    should("not error on strings with integers") {
      checkAll(Arb.int().map { it.toString() }, Arb.currency()) { value, currency ->
      val sut = isMoneyValue(currency)

      val result = sut.validate(value)

      result shouldBe Ok(Money(value, currency))
    }
    }
  }
}
```
Notice how all the lines under `checkAll` have wrong level of indentation. Notice two curly braces one above another. 

## Expected Behavior
Above code should be formatted like this:

(Using IDE formatter also produces the correct results)

```kotlin
class MoneyValidatorsTest : ShouldSpec({
  context("isMoneyValue") {
    should("not error on strings with integers") {
      checkAll(Arb.int().map { it.toString() }, Arb.currency()) { value, currency ->
        val sut = isMoneyValue(currency)

        val result = sut.validate(value)

        result shouldBe Ok(Money(value, currency))
      }
    }
  }
}
```

Note that if I move `Arb.int().map { it.toString() }` expression out of the function call, ktlint formats this correclty:

```kotlin
class MoneyValidatorsTest : ShouldSpec({
  context("isMoneyValue") {
    should("not error on strings with integers") {
      val valueGen = Arb.int().map { it.toString() }
      checkAll(valueGen, Arb.currency()) { value, currency ->
        val sut = isMoneyValue(currency)

        val result = sut.validate(value)

        result shouldBe Ok(Money(value, currency))
      }
    }
  }
}
```

## Your Environment
* Version of ktlint used: `0.42.1`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
* Version of Gradle used (if applicable): `7.0.2`
* Operating System and version: Arch Linux

**Comment by JolandaVerhoef** [comment]
We're seeing the same behavior after upgrading from 0.41.0 to 0.42.1. You can replicate by copying the following:

```
private object Test2 {
    fun test() {
        doSomething({}, 1) {
        // Wrong indentation by Spotless
    }
    }

    fun doSomething(x: () -> Unit, y: Int, z: () -> Unit) {}
}

private object Test3 {
    fun test() {
        doSomething(1) {
            // Indented correctly by Spotless
        }
    }

    fun doSomething(y: Int, z: () -> Unit) {}
}

private object Test4 {
    fun test() {
        doSomething({}) {
            // Indented correctly by Spotless
        }
    }

    fun doSomething(x: () -> Unit, z: () -> Unit) {}
}
```

**Comment by liutikas** [comment]
This is also blocking androidx project from upgrading as it causes a large diff making code harder to read.

**Comment by paul-dingemans** [comment]
Possibly related to https://github.com/pinterest/ktlint/issues/1210

**Comment by paul-dingemans** [comment]
This was fixed by [#1202 ](https://github.com/pinterest/ktlint/pull/1203) and no longer occurs in ktlint 0.43.x

**Comment by romtsn** [comment]
thanks for checking :+1: 

@liutikas just casually checked your config - you can re-enable `import-ordering` as it supports IJ config now
